### ISSUE TITLE
Bug/Controller dismissed on device rotation.

### ISSUE THREAD

**Issue by christoff-1992** [slackhq/PanModal#22]
### Description

The PanModalPresentationController is dismissed when the device is rotated.
This appears to be a result of the dismiss call within the adaptivePresentationStyle call, this is called when the presented view controller becomes horizontally compact.

```SWIFT
public func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
        controller.presentedViewController.dismiss(animated: false, completion: nil)
        return .none
    }
```

However the presented views frame is correctly adjusted in viewWillTransition, but as a result of the above call as soon as this completes the controller is dismissed.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [X] bug

### Requirements (place an `x` in each of the `[ ]`)
* [X] I've read and understood the [Contributing guidelines](https://github.com/slackhq/PanModal/blob/master/CONTRIBUTING.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [X] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible in:

PanModal version: Latest

iOS version: iOS 12.0 but I don't think this will be specific to the issue. (Testing on an iPhone X)

#### Steps to reproduce:

1. Enable device orientation for landscape left / right  
2. Present the controller
3. Rotate the device

#### Expected result:

The views are resized for the new orientation.

#### Actual result:

The presentationTransitionWillBegin is called and as a result the controller is dismissed.

**Comment by TosinAF** [comment]
Hi, 

Thanks for the bug report. Currently we are unable to support fixing non-critical bugs. If you able to submit a PR for this we would be to review.

Cheers,
Tosin

**Comment by christoff-1992** [comment]
> Hi,
> 
> Thanks for the bug report. Currently we are unable to support fixing non-critical bugs. If you able to submit a PR for this we would be to review.
> 
> Cheers,
> Tosin

Thanks @TosinAF - More than happy to look at raising a PR for this but I am unsure on the behaviour that you are looking for. If you have time to discuss the differences between the expected behaviour between iPhones and iPads I can look at resolving the issue.

**Comment by tun57** [comment]
Hey @christoff-1992,

It looks like this line of code is a remnant of previous functionality we had on iPad. We used size classes to determine whether to render a popover or a custom panModal; compact for .custom presentation, regular for .popover, hence the need to dismiss during size class changes. We ended up dropping support for .custom on iPad entirely and just going with the popover (we didn't feel it necessary to build our own custom .popover presentationController to support the transition) but forgot to remove the dismiss.

TDLR; I'll go ahead and remove this, thanks for reporting :)

Stephen
